### ISSUE TITLE
unexpected Throwable (AssertionError)

### ISSUE THREAD

**Issue by GoogleCodeExporter** [typetools/checker-framework#369]
```
What steps will reproduce the problem?
1. Run the Null Checker on the class below
package test;

import static java.util.stream.Collectors.toSet;

import java.util.stream.Stream;

class Test {
    static void test(Stream<Integer> stream) {
        stream.collect(toSet());
    }
}

What is the expected output? What do you see instead?
expected: no errors
actual:error: SourceChecker.typeProcess: unexpected Throwable (AssertionError) 
while processing Test.java; message: PolyCollector: mismatched type kinds: 
@Initialized @NonNull Integer (DECLARED) and ? super @Initialized @NonNull 
Integer extends @Initialized @NonNull Object (WILDCARD); invoke the compiler 
with -AprintErrorStack to see the stack trace.
1 error

What version of the product are you using? On what operating system?
1.8.6/Windows 8/Java 8


```

Original issue reported on code.google.com by `ClovisSe...@gmail.com` on 28 Sep 2014 at 5:26
